source/Ship.cpp

```diff
@@ -2042,10 +2042,16 @@ void Ship::DoGeneration()
 		// close to the star. Carried fighters can't collect fuel or energy this way.
 		if(currentSystem)
 		{
-			double scale = .2 + 1.8 / (.001 * position.Length() + 1);
-			fuel += currentSystem->SolarWind() * .03 * scale * (sqrt(attributes.Get("ramscoop")) + .05 * scale);
+			double ramscoop = max(attributes.Get("ramscoop"), .05);
+			double scale = 0.5 * pow(Mass(), 2. / 3.); // Reusing "scale" instead of making a new temporary variable.
```

Comment:
Might be worth measuring this. `pow` is expensive. Also imo I don't see the value of the comment.

Reply:
The comment was because I believe the standard practice would've been to make one variable for each value that is going to be used later, but I could accomplish this using two variables, which feels better for juggling resources more tightly instead of making a mess and not using a bunch of variables more than once. (It is justification for what might be hard to comprehend if you're just scrolling through the code. ~~It's a bad sign when you feel the need to justify yourself, isn't it?~~)

This should be equivalent to the cube-root of the mass squared, would, say, `val = [cube-root](mass); val *= val;` really be faster than `val = pow(mass, 2/3)`? Otherwise, there is no good cheap approximation afaik... No way to cheaply replicate this with more primitive operations, right?

Reply:
I suppose a crude approximation would be 0.05 * Mass(), but it'll undershoot below 1000 mass, and overshoot above 1000 mass. And at very low mass, you'll lose a lot of ramscoop you would've gotten...